crates/bindings-csharp/BSATN.Runtime/Builtins.cs

```diff
@@ -257,19 +256,49 @@ public sealed record Time(DateTimeOffset Time_) : ScheduleAt;
 
     public static implicit operator ScheduleAt(DateTimeOffset time) => new Time(time);
 
+    /// <summary>
+    /// There are 10 C# Timestamp "Ticks" per microsecond.
+    /// </summary>
+    public static readonly ulong TicksPerMicrosecond = 10;
+
+    public static ulong ToMicroseconds(TimeSpan interval)
+    {
+        return (ulong)interval.Ticks / TicksPerMicrosecond;
+    }
+
+    public static TimeSpan TimeSpanFromMicroseconds(ulong intervalMicros)
+    {
+
+        return TimeSpan.FromTicks((long)(TicksPerMicrosecond * intervalMicros));
+    }
+
+    public static ulong ToMicrosecondsSinceUnixEpoch(DateTimeOffset time)
+    {
+        return ToMicroseconds(time - DateTimeOffset.UnixEpoch);
+    }
+
+    public static DateTimeOffset DateTimeOffsetFromMicrosSinceUnixEpoch(ulong microsSinceUnixEpoch)
+    {
+        return DateTimeOffset.UnixEpoch.AddTicks((long)(TicksPerMicrosecond * microsSinceUnixEpoch));
```

Comment:
Nit: might be slightly cleaner to reuse the `TimeSpan` conversion.

```suggestion
        return DateTimeOffset.UnixEpoch + TimeSpanFromMicroseconds(microsSinceUnixEpoch);
```